app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -1,148 +1,166 @@
+    this._hasScrolled = true
+    return this._hasScrolled
+  }
 
-    self.checkResize = function () {
-      if (_hasResized) {
-        _hasResized = false
-        _hasScrolled = true
-      }
+  HighlightActiveSectionHeading.prototype.checkResize = function () {
+    if (this._hasResized) {
+      this._hasResized = false
+      this._hasScrolled = true
     }
+  }
 
-    self.checkScroll = function () {
-      if (_hasScrolled) {
-        _hasScrolled = false
-        var windowDimensions = self.getWindowDimensions()
-        if (windowDimensions.width <= 768) {
-          self.removeActiveItem()
-        } else {
-          self.updateActiveNavItem()
-        }
+  HighlightActiveSectionHeading.prototype.checkScroll = function () {
+    if (this._hasScrolled) {
+      this._hasScrolled = false
+      var windowDimensions = this.getWindowDimensions()
+      if (windowDimensions.width <= 768) {
+        this.removeActiveItem()
+      } else {
+        this.updateActiveNavItem()
       }
     }
+  }
 
-    self.getAnchors = function () {
-      $.each(self.$anchors, function (i) {
-        var anchorID = $(this).attr('href')
-        // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
-        anchorIDs.push(anchorID)
-      })
+  HighlightActiveSectionHeading.prototype.getWindowDimensions = function () {
+    return {
+      height: window.innerHeight,
+      width: window.innerWidth
     }
+  }
 
-    self.getHeadingPosition = function ($theID) {
-      return $theID.offset()
+  HighlightActiveSectionHeading.prototype.getAnchors = function () {
+    for (var i = 0; i < this.anchors.length; i++) {
+      var anchorID = this.anchors[i].getAttribute('href')
+      // e.g. anchorIDs['#meeting-the-digital-service-standard', '#understand-your-users', '#research-continually']
+      this.anchorIDs.push(anchorID)
     }
+  }
 
-    self.getNextHeadingPosition = function ($theNextID) {
-      return $theNextID.offset()
+  HighlightActiveSectionHeading.prototype.updateActiveNavItem = function () {
+    var windowVerticalPosition = this.getWindowPositions()
+    var footer = document.querySelector('.govuk-footer')
+    if (footer) {
+      var footerPosition = this.getFooterPosition(footer)
     }
 
-    self.getFooterPosition = function ($theID) {
-      return $theID.offset()
-    }
+    for (var i = 0; i < this.anchors.length; i++) {
+      var theID = this.anchorIDs[i]
+      var theNextID = this.anchorIDs[i + 1]
 
-    self.getDistanceBetweenHeadings = function (headingPosition, nextHeadingPosition) {
-      var distanceBetweenHeadings = (nextHeadingPosition - headingPosition)
-      return distanceBetweenHeadings
-    }
+      var $theID = document.getElementById(theID.substring(1)) // remove the # at the start
+      if (theNextID) {
+        var $theNextID = document.getElementById(theNextID.substring(1)) // remove the # at the start
+      }
```

Comment:
This code is a little confusing because it looks like the variable isn't defined unless the if is true (unless you understand all the JS hosting.

I'd suggest using a ternary to make it clearer:

```suggestion
      var $theNextID =  theNextID ? document.getElementById(theNextID) : null
```

Reply:
Thanks, have updated.